File: LeetCode/Kotlin/s1608.kt
class Solution1608 {
    class Solution {
        fun specialArray(nums: IntArray): Int {
            val cnt = nums.fold(IntArray(102)) { r, n -> r[minOf(n, 100)]++; r }
            return (100 downTo 1).find { cnt[it] += cnt[it + 1]; cnt[it] == it } ?: -1
        }
    }
}

File: LeetCode/Rust/s1608.rs
struct Solution1608{}

impl Solution1608 {
    pub fn special_array(nums: Vec<i32>) -> i32 {
        let mut cnt = nums.iter().fold(vec![0;102],|mut r, n|{r[std::cmp::min(*n ,100) as usize] +=1; r});
        (1..101).rev().find(|i|{cnt[*i as usize] += cnt[*i as usize +1]; cnt[*i as usize] == *i}).unwrap_or(-1)
    }
}